advanced/Templates/pihole-FTL.service

```diff
@@ -22,6 +22,9 @@ is_running() {
   return 1
 }
 
+cleanup() {
```

Comment:
Looks good. Do you think this would be a good candidate for a `trap` approach? You have the cleanup function written so all that would need to change is 67-72. 

See https://www.shellscript.sh/trap.html

Reply:
To you want to trap more then just `EXIT`?

Reply:
Trapping for cleanup would be helpful in any condition that isn't `return 0`. I trust your judgement on what you think should be trapped and when the cleanup function should be called.

Reply:
I've implemented a `trap` but after writing the code I doubt if this is a good idea/implementation. We don't want to execute `cleanup` on every exit. E.g.

https://github.com/pi-hole/pi-hole/blob/d80d248ca61d6dde3688050304e5568e86db8bcc/advanced/Templates/pihole-FTL.service#L142-L143

Would delete the files if the service was called with an unknown argument. 
___

Maybe it's simpler and less error-prone if we just execute `cleanup` when we need it and don't trap `EXIT`, but only e.g. `INT`

Reply:
Can you test the previous return code? If it's zero you don't execute the cleanup?

Reply:
But we want the cleanup sometimes with zero exit code (`stop()`).

I think it's getting overly complicated...

Reply:
I changed it to trap only `INT HUP TERM ABRT` (not EXIT) to avoid cleaning files we still need if users only want to get the help text or request the status